### PR TITLE
[Gen 2] Fixes socket leak in UDP.begin()

### PR DESCRIPTION
**Pull request by avtolstoy**
### Problem

`UDP::begin()` can be called when already active with an open socket. It will unconditionally create open a new socket and discard its reference to the old socket while leaving it open.

If this sequence is repeated the system can run out of socket handles preventing new UDP sockets from being opened and even preventing cloud connection.

While users should call `UDP::end()` to close any open socket before a new `UDP::begin()` the implementation should defensively check for and close an open socket on `UDP::begin()`.

**NOTE: this bug only applies to Gen 2 devices**

### Solution

Call `stop()` in `begin()`.

This PR also frees up some flash space in `wiring/no_fixture` test suite for cellular platforms, as we started to overflow.

### Steps to Test

Run `wiring/no_fixture` tests on Gen 2 and Gen 3 platforms. All tests can be excluded (`E` command) except for `MDM_01_socket_writes_with_length_more_than_1023_work_correctly` on cellular platforms and `UDP_begin_does_not_leak_sockets_without_calling_stop` on all the platforms (`i` command to enable the tests selectively).

### Example App

N/A

### References

- [CH27231]

---

### Completeness

- [x] User is totes amazing for contributing!
- [x] Contributor has signed CLA ([Info here](https://github.com/spark/firmware/blob/develop/CONTRIBUTING.md))
- [x] Problem and Solution clearly stated
- [x] Run unit/integration/application tests on device
- [ ] Added documentation
- [ ] Added to CHANGELOG.md after merging (add links to docs and issues)

### CODE REVIEW

File: user/tests/wiring/no_fixture/cellular.cpp [hunk 1]
```diff
@@ -190,16 +205,17 @@ test(MDM_01_socket_writes_with_length_more_than_1023_work_correctly) {
...
 
-    char* responseBuf = new char[2048];
-    memset(responseBuf, 0, 2048);
+    auto responseBuf = std::make_unique<char[]>(bufferSize);
+    assertTrue((bool)responseBuf);
+    memset(responseBuf.get(), 0, bufferSize);
     int responseSize = 0;
     uint32_t mil = millis();
     while(1) {
         while (c.available()) {
```
Comment (eberseth) [inline user/tests/wiring/no_fixture/cellular.cpp]: What to do if responseSize >= bufferSize?

File: user/tests/wiring/no_fixture/cellular.cpp [hunk 2]
```diff
@@ -169,7 +168,16 @@ test(MDM_01_socket_writes_with_length_more_than_1023_work_correctly) {
 #endif // HAL_PLATFORM_NCP
 
     // https://github.com/spark/firmware/issues/1104
-    const char request[] =
+
+    const int dataSize = 1024;
+    const int bufferSize = 2048;
+    auto randData = std::make_unique<char[]>(1024);
```
Comment (eberseth) [inline user/tests/wiring/no_fixture/cellular.cpp]: Use dataSize to initialize instead of magic number.

**Comment by eberseth** [comment]
Can you fix or mute the Codacy/PR review issues?

**Comment by avtolstoy** [comment]
> Can you fix or mute the Codacy/PR review issues?

Just ignore them.

File: user/tests/wiring/no_fixture/cellular.cpp (hunk 1 above)
Comment (avtolstoy) [reply user/tests/wiring/no_fixture/cellular.cpp]: Fixed.

File: user/tests/wiring/no_fixture/cellular.cpp (hunk 2 above)
Comment (avtolstoy) [reply user/tests/wiring/no_fixture/cellular.cpp]: Fixed.
